### PR TITLE
test(#33): normalization edge cases, alias coverage, enharmonic pairs, determinism

### LINKED ISSUE #33: Normalization Edge Cases & Canonical ID Stability
# Issue: Normalization Edge Cases & Canonical ID Stability

## Summary
Expand normalization tests for enharmonics, alias retention, and canonical ID stability under equivalent input forms.

## Why
Normalization integrity is central to deterministic artifacts and reliable retrieval.

## Scope
- Add cases for quality alias variants and root enharmonics.
- Assert canonical ID stability across equivalent representations.
- Assert aliases are retained without silent data loss.
- Confirm deterministic ordering post-normalization.

## Acceptance Criteria
- Added tests cover key edge-case matrix.
- Canonical IDs remain stable and compliant with ADR-0001.
- No alias information is unintentionally dropped.

## Validation Steps
```bash
npm test -- --run test/unit/normalize.test.ts
npm run validate
```

## Follow-ups
- Extend quality map once new chord qualities are introduced.

### PR DESCRIPTION
**Pull request by metal-gogo**
## Summary

Expands `test/unit/normalize.test.ts` to cover the full quality-alias matrix, all enharmonic root pairs, alias deduplication, canonical ID stability across equivalent inputs, and output determinism.

### `test/unit/normalize.test.ts`

**`normalizeQuality` additions (2 new tests):**
- 'maps all quality alias variants in the QUALITY_MAP' — exercises every key: `maj`, `M`, `Δ`, `""`, `min`, `minor`, `maj7`, `major7`, `Δ7`, `m7`, `min7`, `dim`, `diminished`, `m7b5`, `aug`, `augmented`, `sus2`, `sus4`
- 'rejects unsupported aliases' — extended with additional unknown inputs (`unknown`, `maj13`)

**`normalizeRecords` additions (4 new tests):**
1. 'produces the same canonical ID for equivalent quality alias inputs' — all major alias variants yield `chord:C:maj`
2. 'links enharmonic equivalents for all supported enharmonic root pairs' — C#/Db, D#/Eb, F#/Gb, G#/Ab, A#/Bb (10 total pairs)
3. 'deduplicates aliases within a single source record' — ensures `Set` deduplication of repeated alias strings
4. 'produces identical output for identical inputs (determinism)' — JSON.stringify equality across two calls

## Validation

```bash
npm test -- --run test/unit/normalize.test.ts
npm test -- --run
npm run validate
```

## Test counts

Before: 7 tests | After: 12 tests (+5 new)

Closes #33

### CODE REVIEW

File: test/unit/normalize.test.ts [hunk 1]
```diff
@@ -164,4 +200,94 @@ describe("normalizeRecords", () => {
...
+      ["Ab", "G#", "chord:G#:maj"],
+      ["A#", "Bb", "chord:Bb:maj"],
+      ["Bb", "A#", "chord:A#:maj"],
+    ];
+
+    for (const [root, , expectedEquivalent] of pairs) {
+      const records = normalizeRecords([
+        {
+          source: "source-a",
+          url: `https://example.com/${root.toLowerCase()}`,
```
[the remark this answers is not in the document]
Comment (metal-gogo) [reply test/unit/normalize.test.ts]: Applied. Replaced bare root.toLowerCase() URL with a slug using root.replace(/#/g, 'sharp').replace(/b$/, 'flat').toLowerCase() so URLs like 'csharp-major' are used instead of 'c#-major'. This avoids fragment-delimiter ambiguity.

File: test/unit/normalize.test.ts [hunk 2]
```diff
@@ -164,4 +200,94 @@ describe("normalizeRecords", () => {
...
+          root,
+          quality_raw: "major",
+          aliases: [],
+          formula: ["1", "3", "5"],
+          pitch_classes: [],
+          voicings: [{ id: "v1", frets: [null, null, null, null, null, null], base_fret: 1 }]
+        }
+      ]);
+      expect(records[0]?.enharmonic_equivalents, `expected enharmonic for ${root}`).toContain(expectedEquivalent);
+    }
```
[the remark this answers is not in the document]
Comment (metal-gogo) [reply test/unit/normalize.test.ts]: Applied. Added expect(records).toHaveLength(1) before the enharmonic assertion and changed records[0]?.enharmonic_equivalents to records[0].enharmonic_equivalents for direct, type-safe access.

**Comment by metal-gogo** [comment]
Addressed all 3 Copilot review comments in commit 005bc44: (1) URL slugs now use `root.replace(/#/g, 'sharp').replace(/b$/, 'flat').toLowerCase()` to avoid `#` fragment-delimiter issue; (2) added `expect(records).toHaveLength(1)` before enharmonic access for clear failure messages; (3) added `expect(records).toHaveLength(1)` inside each canonical-ID map iteration for same reason.
